File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/entity/Pedido.java
package br.edu.utfpr.serverpedidosjava.entity;

import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;

@Entity
public class Pedido {
    @Id
    @GeneratedValue
    private Integer id;
    @Column(updatable = false)
    private ZonedDateTime data = ZonedDateTime.now();
    @Enumerated(value = EnumType.STRING)
    private StatusPedido status = StatusPedido.AGUARDANDO;
    @ManyToOne(optional = false)
    private Cliente cliente;
    @OneToMany(mappedBy = "id.pedido", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PedidoProduto> produtos = new ArrayList<>();

    public enum StatusPedido {
        AGUARDANDO,
        FATURADO,
        ENVIADO,
        ENTREGUE
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public ZonedDateTime getData() {
        return data;
    }

    public void setData(ZonedDateTime data) {
        this.data = data;
    }

    public StatusPedido getStatus() {
        return status;
    }

    public void setStatus(StatusPedido status) {
        this.status = status;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public List<PedidoProduto> getProdutos() {
        return produtos;
    }

    public void setProdutos(List<PedidoProduto> produtos) {
        this.produtos = produtos;
    }
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/entity/Pedido.kt
package br.edu.utfpr.serverpedidos.entity

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import java.time.ZonedDateTime

@Entity
class Pedido(
    @Id
    @GeneratedValue
    var id: Int = 0,
    @Column(updatable = false)
    var data: ZonedDateTime = ZonedDateTime.now(),
    @Enumerated(value = EnumType.STRING)
    var status: StatusPedido = StatusPedido.AGUARDANDO,
    @ManyToOne(optional = false)
    var cliente: Cliente? = null,
    @OneToMany(mappedBy = "id.pedido", cascade = [CascadeType.ALL], orphanRemoval = true)
    var produtos: List<PedidoProduto> = listOf()
)

enum class StatusPedido {
    AGUARDANDO,
    FATURADO,
    ENVIADO,
    ENTREGUE
}
